src/cpp/client/channel_cc.cc

```diff
@@ -116,7 +117,13 @@ internal::Call Channel::CreateCall(const internal::RpcMethod& method,
     } else if (!host_.empty()) {
       host_str = host_.c_str();
     }
-    grpc_slice method_slice = SliceFromCopiedString(method.name());
+    grpc_slice method_slice;
+    if (static_name != nullptr) {
+      method_slice =
+          grpc_slice_from_static_buffer(static_name, static_name_len);
+    } else {
+      method_slice = SliceFromCopiedString(method.name());
```

Comment:
Once this is submitted, I think it's still worth have a SliceFromArray() for this special case so that we don't create a temp string. Do you think that's a good idea?